crates/ironrdp/examples/server.rs

```diff
@@ -0,0 +1,206 @@
+use rustls::ServerConfig;
+use rustls_pemfile::{certs, pkcs8_private_keys};
+use std::{
+    fs::File,
+    io::BufReader,
+    net::{IpAddr, SocketAddr},
+    num::NonZeroU16,
+    sync::Arc,
+};
+use tokio::time::{sleep, Duration};
+use tokio_rustls::TlsAcceptor;
+
+const HELP: &str = "\
+USAGE:
+  cargo run --example=server -- --host <HOSTNAME> --port <PORT>
+";
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let action = match parse_args() {
+        Ok(action) => action,
+        Err(e) => {
+            println!("{HELP}");
+            return Err(e.context("invalid argument(s)"));
+        }
+    };
+
+    setup_logging()?;
+
+    match action {
+        Action::ShowHelp => {
+            println!("{HELP}");
+            Ok(())
+        }
+        Action::Run { host, port, cert, key } => run(host, port, cert, key).await,
+    }
+}
+
+#[derive(Debug)]
+enum Action {
+    ShowHelp,
+    Run {
+        host: String,
+        port: u16,
+        cert: Option<String>,
+        key: Option<String>,
+    },
+}
+
+fn parse_args() -> anyhow::Result<Action> {
+    let mut args = pico_args::Arguments::from_env();
+
+    let action = if args.contains(["-h", "--help"]) {
+        Action::ShowHelp
+    } else {
+        let host = args.opt_value_from_str("--host")?.unwrap_or(String::from("localhost"));
+        let port = args.opt_value_from_str("--port")?.unwrap_or(3389);
+        let cert = args.opt_value_from_str("--cert")?;
+        let key = args.opt_value_from_str("--key")?;
+        Action::Run { host, port, cert, key }
+    };
+
+    Ok(action)
+}
+
+fn setup_logging() -> anyhow::Result<()> {
+    use tracing::metadata::LevelFilter;
+    use tracing_subscriber::prelude::*;
+    use tracing_subscriber::EnvFilter;
+
+    let fmt_layer = tracing_subscriber::fmt::layer().compact();
+
+    let env_filter = EnvFilter::builder()
+        .with_default_directive(LevelFilter::WARN.into())
+        .with_env_var("IRONRDP_LOG")
+        .from_env_lossy();
+
+    tracing_subscriber::registry()
+        .with(fmt_layer)
+        .with(env_filter)
+        .try_init()
+        .context("failed to set tracing global subscriber")?;
+
+    Ok(())
+}
+
+fn acceptor(cert_path: &str, key_path: &str) -> anyhow::Result<TlsAcceptor> {
+    let cert = certs(&mut BufReader::new(File::open(cert_path)?))?[0].clone();
+    let key = pkcs8_private_keys(&mut BufReader::new(File::open(key_path)?))?[0].clone();
+
+    let mut server_config = ServerConfig::builder()
+        .with_safe_defaults()
+        .with_no_client_auth()
+        .with_single_cert(vec![rustls::Certificate(cert)], rustls::PrivateKey(key))
+        .expect("bad certificate/key");
+
+    // This adds support for the SSLKEYLOGFILE env variable (https://wiki.wireshark.org/TLS#using-the-pre-master-secret)
+    server_config.key_log = Arc::new(rustls::KeyLogFile::new());
+
+    Ok(TlsAcceptor::from(Arc::new(server_config)))
+}
+
+#[derive(Debug)]
+struct InnerHandler {}
```

Comment:
Issue (non-blocking): this appears to be dead code. Strangely enough, the lint is not fired.